File: 프로그래머스/2/118667. 두 큐 합 같게 만들기/두 큐 합 같게 만들기.java
import java.util.*;

class Solution {
    public int solution(int[] queue1, int[] queue2) {
        
        Deque<Integer> que1 = new ArrayDeque<>();
        Deque<Integer> que2 = new ArrayDeque<>();
        
        long sum1 = 0;
        long sum2 = 0;
        
        for(int i=0; i<queue1.length; i++){
            sum1 += queue1[i];
            que1.add(queue1[i]);
            sum2 += queue2[i];
            que2.add(queue2[i]);
        }
        
        if(sum1 == sum2) return 0;
        
        int cnt = 0;
        int size = que1.size()*2+1;
        
        while(cnt <= size && !que1.isEmpty() && !que2.isEmpty()){
            if(sum1 > sum2){
                int out = que1.poll();
                sum1 -= out;
                que2.add(out);
                sum2 += out;
            }else {
                int out = que2.poll();
                sum2 -= out;
                que1.add(out);
                sum1 += out;
            }
            
            cnt++;
            if(sum1 == sum2) return cnt;
        }
        
        
        return -1;
    }
}

File: 프로그래머스/2/118667. 두 큐 합 같게 만들기/두 큐 합 같게 만들기.js
function solution(queue1, queue2) {
    
    var sum1 = queue1.reduce((a,b) => a+b);
    var sum2 = queue2.reduce((a,b) => a+b);
    
    if(sum1 == sum2) return 0;
    
    const total = sum1 + sum2;
    if (total % 2 !== 0) return -1; 
    
    const target = total/2;
    const arr = [...queue1, ...queue2];
    
    let i = 0;
    let j = queue1.length;
    let cnt = 0;
    let size = queue1.length*3;
    
    while(cnt <= size){
        
        if (sum1 === target) return cnt;
        
        if(sum1 > target){
            sum1 -= arr[i];
            i++;
        }else{
            sum1 += arr[j];
            j++;
        }
        
        cnt++;
    }

    return -1;
}